packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -112,4 +125,39 @@ export class EVMSmartWallet {
     public async nfts() {
         return this.crossmintService.fetchNFTs(this.address, this.chain);
     }
+
+    public async sendTransaction<
+        const TAbi extends Abi | readonly unknown[],
+        TFunctionName extends ContractFunctionName<TAbi, "nonpayable" | "payable"> = ContractFunctionName<
+            TAbi,
+            "nonpayable" | "payable"
+        >,
+        TArgs extends ContractFunctionArgs<TAbi, "nonpayable" | "payable", TFunctionName> = ContractFunctionArgs<
+            TAbi,
+            "nonpayable" | "payable",
+            TFunctionName
+        >
+    >({
```

Comment:
I'm thinking we add:

`simulateBeforeSending: boolean = true` -- Simulates the transaction before sending it, and stops it from being sent if simulation failed
`retryConfig: { retryIfFailed: boolean, maxRetryAttempts?: int }` -- an object specifying some retry policy if the transaction failed. I made up the params, we shold put there whatever is relevant

Reply:
I'd argue that if a dev wants tweak any of these they'd be better of using the `viem` client

Reply:
I'd agree with Devlyn here on the first two. The retry settings are specified in `resendCallback`

Reply:
Hmm we can add them later i guess, so nbd

Reply:
Actually what is the confirmation behavior currently? That one does seem worth exposing to me as i dont think there's a one size fits all solution

Reply:
I added confirmation options

Reply:
Why wouldn't someone who cares about that just use the viem client though?

It's only a few LOC more to use and gives them all the options they could ever want.